### PR TITLE
Fix formatting between for and await

### LINKED ISSUE #21084: Whitespace formatting between for and await
**TypeScript Version:**  2.7.0-dev.20180108

**Code**

```ts
async function f() {
    for          await (const x of g()) {
        console.log(x);
    }
}
```

**Expected behavior:**

Should be formatted:

```ts
async function f() {
    for await (const x of g()) {
        console.log(x);
    }
}
```

**Actual behavior:**

No changes are made by the formatter.

### PR DESCRIPTION
**Pull request by esakkiraj**
Fixes #21084

### CODE REVIEW

File: src/services/formatting/rules.ts [hunk 1]
```diff
@@ -321,6 +321,10 @@ namespace ts.formatting {
             rule("NoSpaceAfterCloseBracket", SyntaxKind.CloseBracketToken, anyToken, [isNonJsxSameLineTokenContext, isNotBeforeBlockInFunctionDeclarationContext], RuleAction.Delete),
             rule("SpaceAfterSemicolon", SyntaxKind.SemicolonToken, anyToken, [isNonJsxSameLineTokenContext], RuleAction.Space),
 
+            // Remove extra space between for and await
+            rule("SpaceAfterFor", SyntaxKind.ForKeyword, anyToken, [ isNonJsxSameLineTokenContext ], RuleAction.Space),
+            rule("NoSpaceBeforeAwait", anyToken, SyntaxKind.AwaitKeyword, [ isNonJsxSameLineTokenContext ], RuleAction.Delete),
```
Comment (mhegazy) [inline src/services/formatting/rules.ts]: i do not think this rule is needed.

File: src/services/formatting/rules.ts [hunk 2]
```diff
@@ -321,6 +321,10 @@ namespace ts.formatting {
             rule("NoSpaceAfterCloseBracket", SyntaxKind.CloseBracketToken, anyToken, [isNonJsxSameLineTokenContext, isNotBeforeBlockInFunctionDeclarationContext], RuleAction.Delete),
             rule("SpaceAfterSemicolon", SyntaxKind.SemicolonToken, anyToken, [isNonJsxSameLineTokenContext], RuleAction.Space),
 
+            // Remove extra space between for and await
+            rule("SpaceAfterFor", SyntaxKind.ForKeyword, anyToken, [ isNonJsxSameLineTokenContext ], RuleAction.Space),
```
Comment (mhegazy) [inline src/services/formatting/rules.ts]: call this `SpaceBetweenForAndAwaitKeyword`. also make it between `SyntaxKind.ForKeyword` and `SyntaxKind.AwaitKeyword`, the others are already covered by other rules.

Comment (mhegazy) [inline src/services/formatting/rules.ts]: nit: formatting, please remove space before and after `[` and `]`

**Comment by esakkiraj** [comment]
@mhegazy Update the pull request with the changes mentioned.

**Pull request merged**